sycl/doc/extensions/proposed/sycl_ext_intel_cluster_properties.asciidoc

```diff
@@ -0,0 +1,101 @@
+= sycl_ext_intel_cluster_properties
+
+== Introduction
+
+On FPGA targets, regions of the circuit called clusters may be statically scheduled.
+This extension introduces two kernel properties that specify how statically-scheduled clusters should be implemented for an FPGA target.
+
+== Notice
+
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+== Status
+
+Working Draft
+
+This is a proposed extension specification, intended to gather community
+feedback. Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state. The specification itself may also change in
+incompatible ways before it is finalized. Shipping software products should not
+rely on APIs defined in this specification.
+
+== Version
+
+Last Modified Date: 2022-12-19 +
+Revision: 1
+
+== Contributors
+
+Jessica Davies, Intel +
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6 and
+the following extension:
+
+- link:sycl_ext_oneapi_kernel_properties.asciidoc[sycl_ext_oneapi_kernel_properties]
+
+== Feature Test Macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_INTEL_CLUSTER_PROPERTIES` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version.  Base features are supported.
+|===
+
+== Cluster Properties
+```c++
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+struct stall_free_clusters_key {
+  using value_t = property_value<stall_free_clusters_key>;
+};
+
+struct stall_enable_clusters_key {
+  using value_t = property_value<stall_enable_clusters_key>;
```

Comment:
One could argue that it would be a little more user-friendly to have a `bool` value in properties like this. I.e. the user could do `stall_enable_clusters_key::value_t<SOME_BOOL_CONSTANT>` and switch it on and off at compile time. There are of course ways of doing that without it, but it gets a little more involved.

That said, we have other properties like the current version of this, so I don't think it is critical.

Reply:
Hey Steffen, I moved it over to PR 6348 and there it takes a signed it to signify certain options : 2 of those are on, off controls.